Lib/socket.py

```diff
@@ -650,6 +650,14 @@ def socketpair(family=AF_INET, type=SOCK_STREAM, proto=0):
                 raise
         finally:
             lsock.close()
+
+        # Authenticating avoids using a connection from something else
+        # able to connect to {host}:{port} instead of us.
+        if (
+            ssock.getsockname()[:2] != csock.getpeername()[:2]
+            or csock.getsockname()[:2] != ssock.getpeername()[:2]
+        ):
```

Comment:
Thanks for the review @picnixz! I addressed all your review comments here: https://github.com/python/cpython/pull/122134/commits/fe5da2d40a054b58c58275a77201557d1f8b13ff

On your question wrt scope_id: I think we're safe to not limit ourselves to only host and port, I removed the `[:2]` so that all parts of the address are checked to match.

The platform I'm most interested in whether this test will work with is WASI/WASM? I'm not sure what the socket situation is there.

Reply:
Thanks for the work! I personally have no other comments on the topic and I can't tell you what happens on WASI/WASM since I don't know it myself. We can leave that concern to experts probably (also it will need approval by a core dev so they will probably say something if it's wrong).

Reply:
There are no sockets on WASI, so nothing to worry about there.  I agree that checking all parts of the address tuple makes sense.